pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -868,13 +874,18 @@ func (a *Actuator) requestPowerOff(ctx context.Context, baremetalhost *bmh.BareM
 }
 
 //requestPowerOn removes requestPowerOffAnnotation from baremetalhost which signals BMO to power on the machine
-func (a *Actuator) requestPowerOn(ctx context.Context, baremetalhost *bmh.BareMetalHost) error {
-	if baremetalhost.Annotations == nil {
-		baremetalhost.Annotations = make(map[string]string)
+func (a *Actuator) requestPowerOn(ctx context.Context, machine *machinev1beta1.Machine, baremetalhost *bmh.BareMetalHost) error {
+	if machine.Annotations == nil {
+		machine.Annotations = make(map[string]string)
 	}
+	machine.Annotations[powerOnRequestTimestampAnnotation] = time.Now().Format(annotationTimestampFormat)
```

Comment:
It's not technically necessary to add yet another annotation here; we could just add the data to the existing requestPowerOffAnnotation IIUC. I guess this is more descriptive though.

Reply:
Exactly, it just makes it all a lot more clear.